src/LayerPlan.cpp

```diff
@@ -774,6 +774,31 @@ void LayerPlan::writeGCode(GCodeExport& gcode)
                 }
                 gcode.writeTemperatureCommand(prev_extruder, prev_extruder_temp, wait);
             }
+
+            if (!getPrimeTowerIsPlanned())
```

Comment:
No, there is a special extruder-change retract which is 16mm.
If you look carefully, each time we switch from T0 to T1 we retract 16mm in total compared to the last extrusion E value and that is also the amount primed before the first extrusion move when we switch back to that extruder.

I don't see a problem when looking at the gcode you posted named 'UM3_SpiralMultiMaterial_5.gcode`...

If there still is a bug with unretraction or with the prime poop then that problem should be solved properly by fixing the unretraction and/or prime poop implementation, not by introducing a hotfix to solve it only for spiralize mode.